packages/testsweets/lib/src/models/application_models.dart

```diff
@@ -15,47 +15,37 @@ class WidgetDescription with _$WidgetDescription {
     String? id,
 
     /// The name of the view this widget was captured on
-    required String viewName,
+    String? viewName,
```

Comment:
The same change here. I don't think it's good to make it nullable. The viewname should be present whenever a view is captured. Otherwise there's an error and we don't want the user to use the package in that state.

Reply:
The reason behind this is the view widget don't need a viewName and name
so I made the viewName null 
It's breaking the interface segregation principal
Can you give me another suggestion to this problem?